vector-stores/spring-ai-azure-store/src/test/java/org/springframework/ai/vectorstore/azure/AzureVectorStoreIT.java

```diff
@@ -300,6 +301,14 @@ public static class Config {
 
 		@Bean
 		public SearchIndexClient searchIndexClient() {
+			// Only set AZURE_AI_SEARCH_TEST_KEYLESS if role-based authentication is set up correctly on the integration service
+			// https://learn.microsoft.com/azure/search/search-security-rbac
+			if (System.getenv("AZURE_AI_SEARCH_TEST_KEYLESS").equals("true")) {
```

Comment:
I believe we should add some sort of `useKeyless` or similar name to `AzureVectorStoreProperties` rather than hard coding a search for an environment variable.  Boot will be able to support getting this value from an env-var as well as lots of other places.